lib/templates.py

```diff
@@ -83,18 +86,24 @@ class BasicTextTemplate(Script):
     def __init__(self, **kwargs):
         Script.__init__(self, **kwargs)
 
-    def download(self, engine=None, debug=False):
-        Script.download(self, engine, debug)
+    def download(self, engine=None, debug=False, populate=False):
+        Script.download(self, engine, debug, populate)
 
         for key in self.urls.keys():
             if key not in self.tables.keys():
                 self.tables[key] = Table(key, cleanup=Cleanup(correct_invalid_value,
                                                               nulls=[-999]))
 
         for key, value in self.urls.items():
-            self.engine.auto_create_table(self.tables[key], url=value)
-            self.engine.insert_data_from_url(value)
-            self.tables[key].record_id = 0
+            if not populate:
+                self.engine.auto_create_table(self.tables[key], url=value)
+                self.engine.insert_data_from_url(value)
+                self.tables[key].record_id = 0
+            else:
+                self.engine.auto_create_table(self.tables[key], url=value, populate=True)
```

Comment:
Since `populate` is passed in to this function, this line and 99 can be run as a single line before the conditional as:

```
self.engine.auto_create_table(self.tables[key], url=value, populate=populate)
```